### PR TITLE
DACT-424 Date of removal patch and API validation

### PR DESCRIPTION
**Pull request by jditchburn**
When the user enters the resignation date and clicks continue:
- Fetch the current etag from company appointments
- Patch the etag and removal date into the filing
- Run api validation on the filing and display errors

[DACT-424](https://companieshouse.atlassian.net/browse/DACT-424)

[DACT-424]: https://companieshouse.atlassian.net/browse/DACT-424?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/controllers/remove.director.controller.ts [hunk 1]
```diff
@@ -27,9 +37,47 @@ export const post = async (req: Request, res: Response, next: NextFunction) => {
   try {
     //Hardcoded for now
     req.params[urlParams.PARAM_SUBMISSION_ID] = "645d1188c794645afe15f5cc";
+
+    const transactionId = urlUtils.getTransactionIdFromRequestParams(req);
+    const submissionId = "645d1188c794645afe15f5cc";
```
Comment (okoye-companieshouse) [inline src/controllers/remove.director.controller.ts]: Value could have been made into a constant for easy reuse

**Review by jbestwickch — APPROVED** [review]

File: src/controllers/remove.director.controller.ts [hunk 2]
```diff
@@ -27,9 +37,47 @@ export const post = async (req: Request, res: Response, next: NextFunction) => {
...
+
     const nextPageUrl = urlUtils.getUrlToPath(REMOVE_DIRECTOR_CHECK_ANSWERS_PATH, req);
     return res.redirect(nextPageUrl);
   } catch (e) {
     return next(e);
   }
 };
+
+export function formatValidationError(errorMessage: string): FormattedValidationErrors {
+  const errors = { errorMessage: [] } as any;
```
Comment (okoye-companieshouse) [inline src/controllers/remove.director.controller.ts]: Maybe Record<string, any>

File: src/controllers/remove.director.controller.ts (hunk 1 above)
Comment (jditchburn) [reply src/controllers/remove.director.controller.ts]: This is from an old commit that is no longer in the code

**Review by okoye-companieshouse — APPROVED** [review]

File: src/controllers/remove.director.controller.ts (hunk 2 above)
Comment (jditchburn) [reply src/controllers/remove.director.controller.ts]: The data structure has to be consistent with existing error validation so an array must be used here

Comment (okoye-companieshouse) [reply src/controllers/remove.director.controller.ts]: However, at the bottom we add a new key which accepts value string. Hence, transforming the the datatype Record<string, Array<string>> to Record<string, any>. Without that errors = 9 will also be acceptable.

**Pull request merged**